src/keys/bip39.rs

```diff
@@ -43,17 +165,43 @@ pub mod wordlist {
     #[cfg_attr(docsrs, doc(cfg(feature = "bip39-jp")))]
     include!("bip39.jp.rs");
 
-    #[derive(Debug, PartialEq, Eq)]
-    pub enum Error {
-        InvalidEntropyCount(usize),
-        NoSuchWord(String),
-        ChecksumMismatch,
-    }
-
     const fn cs(ent: usize) -> usize {
         ent / 32
     }
 
+    impl<'a> Wordlist<'a> {
+        // TODO: should it be pub?
+        const fn new_unchecked(separator: char, words: &'a [&'a str; 2048]) -> Self {
+            Self { words, separator }
+        }
+
+        pub fn new(separator: char, words: &'a [&'a str; 2048]) -> Result<Self, Error> {
+            // normalize separator char
+            let mut s = String::new();
+            s.push(separator);
+            let s: String = s.nfkd().collect();
+            let mut s_chars = s.chars();
+
+            if let Some(separator) = s_chars.next() {
+                if s_chars.next().is_none() {
+                    // each word is normalized and without separator
+                    words.iter().try_for_each(|word| {
+                        if is_nfkd(word) && !word.contains(separator) {
+                            Ok(())
+                        } else {
+                            Err(Error::BadWordlist)
+                        }
+                    })?;
+                    Ok(Self { words, separator })
+                } else {
+                    Err(Error::BadSeparator)
```

Comment:
This erros says, that separator is valid only when it can be normalized to one character.
But for instance `Å` is being normalized to two charcters and wont be valid in this case. Is this expected behavior?

Reply:
Yes. I think one-char separator covers most real-world use-cases (eg. space, tab, new line, even this weird japanese space normalizes to the ASCII space). If we need to support multi-char separator then this definitely changes matters.

Reply:
Done.